File: code-examples/cpp/e06/src/content/Logarithm.cpp
#include "../../include/content/Logarithm.h"
#include "cmath"

// Before refactor
double Logarithm::doubleValue() {
  return log(this->argument) / log(this->base);
}
Logarithm::Logarithm(double inputBase, double inputArgument) {
  this->base = inputBase;
  this->argument = inputArgument;
}

File: code-examples/java/e06/src/pl/poznan/put/log/Logarithm.java
package pl.poznan.put.log;

// Before refactor
public class Logarithm implements Number {

  private double base, argument;

  @Override
  public double doubleValue() {
    return Math.log(this.argument) / Math.log(this.base);
  }

  public Logarithm(double inputBase, double inputArgument) {
    this.base = inputBase;
    this.argument = inputArgument;
  }
}
